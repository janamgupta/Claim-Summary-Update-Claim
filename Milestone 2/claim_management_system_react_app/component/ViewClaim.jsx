/* eslint-disable react/prop-types */
import React from 'react';
import Container from 'react-bootstrap/Container';
import { Card, ListGroup, ListGroupItem, CardColumns, Button } from 'react-bootstrap';
import axios from 'axios';
import { browserHistory } from 'react-router';
import AppRouter from './AppRoute.jsx';
import { Logger } from 'react-logger-lib';
import { store } from '../reducers/store.js';
import { claimList } from '../reducers';
import { connect } from 'react-redux';
import { faEdit, faIdBadge, faUser, faAddressCard, faCheckSquare, faCalendarCheck, faCalendarTimes } from "@fortawesome/free-regular-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faList } from '@fortawesome/free-solid-svg-icons';

class ViewClaim extends React.Component {
    constructor(props) {

        super(props);
        this.state = {
            claims: [],
            userName: ''
        }
    }

    componentDidMount() {
        // let userName = localStorage.getItem('username');
        // this.setState({ userName });
        axios.get(`http://localhost:3000/api/viewClaims`)
            .then(res => {
                const claims = res.data;
                this.props.claimList(claims);
            });
            store.subscribe(() => {
                this.setState(
                    { claims: store.getState() }
                )
            });
    }

    updatePage(claim) {
        window.event.preventDefault();
        Logger.of('App.ViewClaim.updatePage').info('claim data', claim);
        browserHistory.push("updateClaim/" + claim.Emp_id);
    }
    render() {
        return (

            <div>
                <AppRouter username={this.state.userName}></AppRouter>
                <Container>
                    <h3 className="text-center mt-3"><FontAwesomeIcon icon={faList}/>  Claim Summary</h3>
                    <CardColumns>
                        {this.state.claims.map((claim, index) =>

                            <Card key={index} className="text-center boxes">
                                <Card.Header>Claim Summary</Card.Header>
                                <Card.Body>
                                <ListGroup className="list-group-flush">
                                    <ListGroupItem><span><FontAwesomeIcon icon={faIdBadge}/> Employee ID : </span><span>{claim.Emp_id}</span></ListGroupItem>
                                    <ListGroupItem><span><FontAwesomeIcon icon={faUser}/> Employee Name : </span>{claim.Employee_Name}</ListGroupItem>
                                    <ListGroupItem><span><FontAwesomeIcon icon={faAddressCard}/> Claim Number : </span>{claim.Claim_Number}</ListGroupItem>
                                    <ListGroupItem><span><FontAwesomeIcon icon={faCheckSquare}/> Claim Program : </span>{claim.Claim_Program}</ListGroupItem>
                                    <ListGroupItem><span><FontAwesomeIcon icon={faEdit}/> Claim Type : </span>{claim.Claim_Type}</ListGroupItem>
                                    <ListGroupItem><span><FontAwesomeIcon icon={faCalendarCheck}/> Claim Start Date : </span>{claim.Claim_Start_Date}</ListGroupItem>
                                    <ListGroupItem><span><FontAwesomeIcon icon={faCalendarTimes}/> Claim End Date : </span>{claim.Claim_End_Date}</ListGroupItem>
                                </ListGroup>
                                <Button variant="outline-primary" onClick={()=> this.updatePage(claim)}>Update <FontAwesomeIcon icon={faEdit}/></Button>
                                </Card.Body>
                            </Card>
                        )
                        }
                    </CardColumns>
                </Container>
            </div>
        )
    }
}

const mapStateToProps = state => {
    return {
        claimList: state.claims
    }
}

const mapDispatchToProps = dispatch => {
    return {
        claimList: claims => dispatch(claimList(claims))
    }
}
export default connect(
    mapStateToProps,
    mapDispatchToProps)(ViewClaim)
