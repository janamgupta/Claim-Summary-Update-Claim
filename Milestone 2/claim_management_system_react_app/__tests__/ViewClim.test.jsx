/* eslint-disable no-unused-vars */
/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import ViewClaim from '../component/ViewClaim';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

const mockStore = configureMockStore();
const store = mockStore({});
describe('View Claims', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<Provider store={store}>
            <ViewClaim />
        </Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

// test('should fetch users', () => {
//     const users = [{"_id":1,"username":"abc12","password":"abc12"},{"_id":2,"username":"pqr12","password":"pqr12"}];
//     const resp = {data: users};
//     Axios.get.mockResolvedValue(resp);
//     let login = new Login();  
//     login.userValidation().then(data => expect(data).toEqual(users));
//   });

