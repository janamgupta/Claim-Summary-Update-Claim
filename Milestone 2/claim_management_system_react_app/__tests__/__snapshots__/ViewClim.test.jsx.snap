// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Claims renders correctly 1`] = `
<div>
  <div>
    <nav
      className="routeNav navbar navbar-expand-md navbar-light bg-primary"
    >
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          className="nav-item"
        >
          <a
            className="nav-link"
            onClick={[Function]}
            style={Object {}}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-home fa-w-18 "
              data-icon="home"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             Home
          </a>
        </div>
        <div
          className="nav-item"
        >
          <a
            className="nav-link"
            onClick={[Function]}
            style={Object {}}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-list fa-w-16 "
              data-icon="list"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M80 368H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm0-320H16A16 16 0 0 0 0 64v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16V64a16 16 0 0 0-16-16zm0 160H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm416 176H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             View Claim
          </a>
        </div>
        <div
          className="nav-item"
        >
          <a
            className="nav-link"
            onClick={[Function]}
            style={Object {}}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-address-card fa-w-18 "
              data-icon="address-card"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M528 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h480c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-352 96c35.3 0 64 28.7 64 64s-28.7 64-64 64-64-28.7-64-64 28.7-64 64-64zm112 236.8c0 10.6-10 19.2-22.4 19.2H86.4C74 384 64 375.4 64 364.8v-19.2c0-31.8 30.1-57.6 67.2-57.6h5c12.3 5.1 25.7 8 39.8 8s27.6-2.9 39.8-8h5c37.1 0 67.2 25.8 67.2 57.6v19.2zM512 312c0 4.4-3.6 8-8 8H360c-4.4 0-8-3.6-8-8v-16c0-4.4 3.6-8 8-8h144c4.4 0 8 3.6 8 8v16zm0-64c0 4.4-3.6 8-8 8H360c-4.4 0-8-3.6-8-8v-16c0-4.4 3.6-8 8-8h144c4.4 0 8 3.6 8 8v16zm0-64c0 4.4-3.6 8-8 8H360c-4.4 0-8-3.6-8-8v-16c0-4.4 3.6-8 8-8h144c4.4 0 8 3.6 8 8v16z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             Contact Us
          </a>
        </div>
        <div
          className="nav-item"
        >
          <a
            className="nav-link"
            onClick={[Function]}
            style={Object {}}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-info-circle fa-w-16 "
              data-icon="info-circle"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             About
          </a>
        </div>
        <div
          className="ml-auto user mr-5 nav-item"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-user-circle fa-w-16 fa-lg "
            data-icon="user-circle"
            data-prefix="far"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 496 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M248 104c-53 0-96 43-96 96s43 96 96 96 96-43 96-96-43-96-96-96zm0 144c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm0-240C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-49.7 0-95.1-18.3-130.1-48.4 14.9-23 40.4-38.6 69.6-39.5 20.8 6.4 40.6 9.6 60.5 9.6s39.7-3.1 60.5-9.6c29.2 1 54.7 16.5 69.6 39.5-35 30.1-80.4 48.4-130.1 48.4zm162.7-84.1c-24.4-31.4-62.1-51.9-105.1-51.9-10.2 0-26 9.6-57.6 9.6-31.5 0-47.4-9.6-57.6-9.6-42.9 0-80.6 20.5-105.1 51.9C61.9 339.2 48 299.2 48 256c0-110.3 89.7-200 200-200s200 89.7 200 200c0 43.2-13.9 83.2-37.3 115.9z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
            
          
        </div>
        <div
          className="nav-item"
        >
          <button
            className="btn btn-outline-light"
            onClick={[Function]}
          >
            signout   
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-sign-out-alt fa-w-16 "
              data-icon="sign-out-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </div>
      </div>
    </nav>
  </div>
  <div
    className="container"
  >
    <h3
      className="text-center mt-3"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-list fa-w-16 "
        data-icon="list"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M80 368H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm0-320H16A16 16 0 0 0 0 64v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16V64a16 16 0 0 0-16-16zm0 160H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm416 176H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
        Claim Summary
    </h3>
    <div
      className="card-columns"
    />
  </div>
</div>
`;
